src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java

```diff
@@ -209,36 +212,52 @@ private String getCorrectAnswer(String questionId, DocumentReference questionDoc
       // TODO Auto-generated catch block
       e.printStackTrace();
     }
-    
 
     return correctAnswerId;
   }
 
-  private void addPoints(String userId, DocumentReference gameInstanceDocRef, Firestore firestoreDb) {
+  private void updateStudentStats(boolean isAnswerCorrect, String userId, DocumentReference gameInstanceDocRef,
+      Firestore firestoreDb) {
     DocumentReference userInGameInstanceDocRef = gameInstanceDocRef.collection("students").document(userId);
 
     firestoreDb.runTransaction(transaction -> {
-      // retrieve document and increment population field
       DocumentSnapshot snapshot = transaction.get(userInGameInstanceDocRef).get();
-      long oldPoints = snapshot.getLong("points");
-      transaction.update(userInGameInstanceDocRef, "points", oldPoints + 10);
+
+      long oldNumberAnswered = snapshot.getLong("numberAnswered");
+      transaction.update(userInGameInstanceDocRef, "numberAnswered", oldNumberAnswered + 1);
```

Comment:
I think we will need to mke a function in the Answer dao for this updates and in each update get the actual stats and update, like a getAnswer() and then updateAnswer(Answer answer). If we have this functions in the dao this servlet will be more testable.